### PR TITLE
Added node class to tagged metrics

### PR DESCRIPTION
**Pull request by burdandrei**
For easier monitoring nomad cluster parts

### CODE REVIEW

**Comment by jippi** [comment]
nice!

**Comment by burdandrei** [comment]
@jippi  hope it would be nice -19 ;)

File: client/client.go [hunk 1]
```diff
@@ -1870,7 +1870,15 @@ DISCOLOOP:
 func (c *Client) emitStats() {
 	// Assign labels directly before emitting stats so the information expected
 	// is ready
-	c.baseLabels = []metrics.Label{{Name: "node_id", Value: c.NodeID()}, {Name: "datacenter", Value: c.Datacenter()}}
+	var emittedNodeClass string
+	if emittedNodeClass = c.config.Node.NodeClass; emittedNodeClass == "" {
+		emittedNodeClass = "none"
+	}
+	c.baseLabels = []metrics.Label{
```
Comment (dadgar) [inline client/client.go]: Can you update to

```
c.baseLabels = []metrics.Label{{Name: "node_id", Value: c.NodeID()}, {Name: "datacenter", Value: c.Datacenter()}}
if node := c.Node(); node.NodeClass != "" {
  c.baseLabels = append(c.baseLabels, {Name: "node_class", Value: node.NodeClass})
}
```

Comment (burdandrei) [reply client/client.go]: sure! original idea was to add "none" like it's listed in CLI `nomad node-status`

**Comment by burdandrei** [comment]
@dadgar I rebased on master and  made as you asked, but the build is failing on https://github.com/hashicorp/nomad/blob/master/nomad/structs/structs.go#L2966
@preetapan you want me to look at this?

File: client/client.go (hunk 1 above)
Comment (jippi) [reply client/client.go]: I like `none` idea - having a sparse field for `class` will be confusing :)

**Comment by burdandrei** [comment]
we will not add the flag to report "none" client class, right?

File: client/client.go (hunk 1 above)
Comment (burdandrei) [reply client/client.go]: we will not add the flag to report "none" client class, right?

Comment (dadgar) [reply client/client.go]: Problem is all possible strings you could add are going to be valid possible node classes

Comment (burdandrei) [reply client/client.go]: @dadgar while working on nomad monitoring tutorial I ran into a problem with grouping nodes by class when node class is not emitted from not nodes with no class configured, so I reverted the code to my previous version when none is emitted from not tagged nodes. 
The issue of all possible strings are going to be valid can be addressed on config parsing stage.
Let me know if you want me to add some restrictions there.

**Pull request merged**